### ISSUE TITLE
Style/WordArray cop would be more inteligent in case of multidimensional array

### ISSUE THREAD

**Issue by deepj** [rubocop/rubocop#10194]
I have some cases of multidimensional array where (from my perspective) default Style/WordArray cop makes the code less readable and confusing. See this example:

```ruby
CONTENT_TYPE = [
    ['TSR', 'Teaser'],
    ['PRO', 'Promo'],
    ['RTG', 'Rating Tag'],
    ['RTG-F', 'Rating Tag for Feature'],
]
```

Rubocop forces me to change it to:

```ruby
CONTENT_TYPE = [
    %w[TSR Teaser],
    %w[PRO Promo],
    ['RTG', 'Rating Tag'],
    ['RTG-F', 'Rating Tag for Feature'],
]
```

I don't think this helps to better readability. I would wish if this would be valid and fine for default Style/WordArray cop

```
CONTENT_TYPE = [
    ['TSR', 'Teaser'],
    ['PRO', 'Promo'],
    ['RTG', 'Rating Tag'],
    ['RTG-F', 'Rating Tag for Feature'],
]
```

Please, consider this as an example ^^.

**Comment by tejasbubane** [comment]
I would argue that it looks confusing because your arrays happen to have some strings with spaces (hence cannot be converted to WordArray. Had it been as follows, it is not confusing anymore:
```ruby
CONTENT_TYPE = [
    %w[TSR Teaser],
    %w[PRO Promo]
]
```

However one solution could be to add a configuration flag which disable cop for nested arrays:
```ruby
# EnabledForNestedArrays: false
# good
CONTENT_TYPE = [
    ['TSR', 'Teaser'],
    ['PRO', 'Promo']
]

# EnabledForNestedArrays: true
# bad
CONTENT_TYPE = [
    ['TSR', 'Teaser'],
    ['PRO', 'Promo']
]
```
Although I don't prefer adding more complexity with flags, I can put together a PR if maintainers are willing to accept this.

**Comment by deepj** [comment]
@tejasbubane I appreciate your comment but the arrays cannot be changed. 

```ruby
CONTENT_TYPE = [
    ['TSR', 'Teaser'],
    ['PRO', 'Promo'],
    ['RTG', 'Rating Tag'],
    ['RTG-F', 'Rating Tag for Feature'],
]
```
What I need in this case the cop ignore kind of arrays. I think your suggestion to turn off it for them is one of the solution. 

@koic What do you think?

**Comment by deepj** [comment]
@fatkodima Thank you!
